zio-json/shared/src/main/scala/zio/json/internal/readers.scala

```diff
@@ -161,7 +161,7 @@ final class WithRetractReader(in: java.io.Reader) extends RetractReader with Aut
 private[zio] sealed trait RecordingReader extends RetractReader {
   def rewind(): Unit
 }
-private[zio] object RecordingReader {
+object RecordingReader {
   def apply(in: OneCharReader): RecordingReader =
```

Comment:
The macro basically inlines this code. Therefore we get a compile time error if it is package private.